src/traces/image/attributes.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var plotAttrs = require('../../plots/attributes');
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'A 2-dimensional array in which each element is an array of 3 or 4 numbers representing a color.',
+        ].join(' ')
+    },
+    colormodel: {
+        valType: 'enumerated',
+        values: ['rgb', 'rgba', 'hsl', 'hsla'],
+        dflt: 'rgb',
+        role: 'info',
+        editType: 'plot',
+        description: 'Color model used to map the numerical color components described in `z` into colors.'
+    },
+    zmin: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Array defining the lower bound for each color component.',
+            'For example, for the `rgba` colormodel, the default value is [0, 0, 0, 0].'
+        ].join(' ')
+    },
+    zmax: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Array defining the higher bound for each color component.',
+            'For example, for the `rgba` colormodel, the default value is [255, 255, 255, 1].'
+        ].join(' ')
+    },
+    x0: {
```

Comment:
So no `x` and `y` for this iteration? Is that intended?

Reply:
> Oh and one more thing @antoinerg: I didn't notice any funky CSS tricks like you had in one of your proof-of-concepts. I guess you decide to not go down that route, can you confirm?

I confirm that I :zap: the CSS trick I had in my proof of concept. If you want me to, I could add a few comments/references about CSS `image-rendering: pixelated` which could potentially be used in the future to speed up rendering on browsers that support it (almost all of them except Safari and potentially SVG editors like Illustrator/Inkscape).

Reply:
As far as I know, this wasn't a requirement for that iteration so I did not include it. For your information, most other libraries do not support this either. It is usually assumed that an image is made of pixels of identical size arranged in a regular fashion as in a square crystal lattice. I'd be surprised if users ask for `x` and `y` but we'll see :smile:

Reply:
`x0` and `dx` seem enough for a v1... any obvious cases where that wouldn't be enough? @jonmmease @emmanuelle @alexcjohnson ?

Reply:
FWIW I'm also very much ok with not implementing `x` and `y` in this iteration.